pandora-common/src/character/assetPreferences.ts

```diff
@@ -129,20 +131,6 @@ export function CleanupAssetPreferences(assetManager: AssetManager, {
 			case 'lock':
 				break;
 		}
-		if (!allowBaseState && KnownObject.keys(value).length === 1 && value.base === 'normal') {
-			let hasAttribute = false;
-			for (const attribute of asset.staticAttributes) {
-				if (attributes[attribute] != null) {
-					hasAttribute = true;
-					break;
-				}
-			}
-			if (!hasAttribute) {
-				delete assets[key];
-				hasInvalid = true;
-				continue;
-			}
-		}
```

Comment:
I am not sure if it is a good idea to remove this, assets that do not have any attribute affecting them and are in default state have no effect and should be cleaned if possible

The only reason to allow them is to be user-friendly so you can put a currently worn item in "normal" and then change an attribute to "doNotRender" which would not be possible otherwise

and the same for attributes unless we extend the base option with other thing "normal" should be dropped and not stored in database, ideally we should try to stay as sparse as we can

Reply:
The reason is to preserve the item's setting - not changing unexpectedly.
Imagine the following flow:
- User sets some attribute to "Maybe"
- One items blocked by it is fully okay for them, however, so they override that item to "Normal"
- *) Now they go the attribute, wanting to change it to "Prevent", but they accidentally (or temporarily intentionally) change it to "Normal"
- Now change it correctly to "Prevent"
- ... but the item is now set as "Prevent", despite them having set it manually to "Normal"

To make things even more strange this behaviour is not consistent if you insert server restart/shard switch after the `*)` point.
Also instead of this example of `*)` point you can imagine asset definition changing to not have that attribute and later the attribute being re-added - which is even worse as user's preference was lost for them seemingly randomly.

Reply:
Oh, and for attributes: For attributes I drop it always (both in transit and in database) - I did that by changing the update call to contain full data, so lack of entry means resetting it to the default value.

Reply:
Well, I know this cleanup can cause slightly unexpected behavior.

I would ignore assets repo changes since that can be viewed as unexpected regardless since the user may not be aware what attribute changed for items.

So, with that out of the way, step 3 only causes issues if and only if the server restarts, which I do not see as that big of a problem, though delaying it more could be an option to only happen at a fixed time after a change has been made (maybe a week since last change, the user surely won't remember any of it)

But if you feel this data is not worth the hassle to cleanup and should not effect effect size in a meaningful way feel free to close this thread